src/Controls/src/Xaml/XamlServiceProvider.cs

```diff
@@ -27,6 +27,9 @@ internal XamlServiceProvider(INode node, HydrationContext context)
 				IXmlLineInfoProvider = new XmlLineInfoProvider(xmlLineInfo);
 
 			IValueConverterProvider = new ValueConverterProvider();
+
+			if (node is IElementNode elementNode)
+				Add(typeof(IXamlDataTypeProvider), new XamlDataTypeProvider(elementNode));
```

Comment:
Will this service be added even when XamlC creates the new XamlServiceProvider instance? And if it is, won't this be added for almost any extension, even if `IXamlDataTypeProvider` is not in its `[RequiredService([...])]`?

Reply:
yes, but I think I have a PR that only add the services that are required

Reply:
going to fix this